tutorials/Exceptions.md

```diff
@@ -0,0 +1,209 @@
+
+```haskell
+-- | Throws an exception of type e.
+throw :: Exception e => e -> m a
+
+-- | If the computation in the first argument throws an error, apply the second
+-- argument to it as a fallback measure.
+catch :: Exception e => m a	-> (e -> m a)	-> m a
+
+-- | Runs the computation in the first argument, and then the computation in the
+-- second one afterwards regardless of an error was thrown.
+-- | finally is only required if you want to ensure some computation (m b)
+-- always happens after the computation (m a). For the most part you will only
+-- need throw and catch.
+finally :: m a -> m b -> m a
+```
+
+You may notice that `throw` and `catch` are aliased as `throwBot` and `catchBot` respectively. These aliases do exactly the same thing, but specify that the type of exception is always `BotException`. This is very useful when writing fallbacks that the compiler cannot determine the type of. Please use `throwBot` and `catchBot` instead of their original counterparts.
+
+### Example
+
+Here is an example of a function `foo` that does nothing but throw a `GenericException`. It is called by the function `bar` which catches the exception and responds with a fallback:
+
+```haskell
+foo = throwBot $ GenericException "TutorialException" "This is the description of the exception."
+
+bar = foo `catchBot` \e -> putStrLn "An exception was caught!"
+```
+
+If you evaluate `foo` by itself you'll get the default output for when an uncaught error occurs:
+`*** Exception: GenericException "TutorialException" "This is the description of the exception."`
+
+Note that this will only appear in the command line, so Discord users won't see it but only a lack of response from the bot.
+
+If you evaluate `bar` it will catch the exception from `foo` and output `An exception was caught!`.
+
+You won't always have to worry about catching exceptions. Any exception that happens while evaluating a command response will be caught in `Tablebot/Handler.hs` and formatted into a nice embed message for Discord users to see. Times when you might want to catch exceptions is when you don't want the command to fail if there's an error. You might also want to clarify exactly what the error was. A good example of this is with functions involving random number generation.
+
+The following example works using the `chooseOne` function from `Tablebot/Plugin/Random.hs`. Note its sibling function `chooseOneWithDefault`, which uses a `catch` to ensure it never propagates an error. The intention with the code below is that the user provides a minimum and maximum number and the bot responds with a random number inclusively between them:
+
+```haskell
+randomNumber :: Command
+randomNumber =
+  Command
+    "randomNumber"
+    ( parseComm $ \m -> do
+        low <- number
+        space
+        high <- number
+        x <- chooseOne [low..high]
+        sendMessage m $ pack $ show x
+    )
```

Comment:
does this work? it seems like you're merging parsing and commands here - perhaps use a smart command to get the numbers instead?